### PR TITLE
NestedFolderPicker: separate toggle to force enable picker without `nestedFolders`

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- adds a new toggle (`newFolderPicker`) to force the nested folder picker to be enabled even when `nestedFolders` isn't enabled

**Why do we need this feature?**

- so we can enable the nested folder picker without `nestedFolders` also being enabled

**Who is this feature for?**

- everyone

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/featuremgmt/registry.go [hunk 1]
```diff
@@ -1323,5 +1323,13 @@ var (
 			AllowSelfServe: false,
 			Created:        time.Date(2024, time.January, 10, 12, 0, 0, 0, time.UTC),
 		},
+		{
+			Name:         "nestedFolderPickerOverride",
```
Comment (ashharrison90) [inline pkg/services/featuremgmt/registry.go:1327]: naming suggestions welcome - i have no ideas 😂

Comment (joshhunt) [reply pkg/services/featuremgmt/registry.go:1327]: what about just `newFolderPicker`?

Comment (joshhunt) [reply pkg/services/featuremgmt/registry.go:1327]: either way, it doesn't really matter that much I think if we don't intend for anyone to turn this on.

**Comment by joshhunt** [comment]
Pushed up a change to reduce the leading spacing before each item when the folder open chevrons aren't there 

Screenshot 2024-01-15 at 11 00 10 am

**Review by joshhunt — APPROVED** [review]
Tested a bunch of scenarios - LGTM!

**Pull request merged**